eng/common/scripts/get-codeowners.ps1

```diff
@@ -1,43 +1,113 @@
 param (
   $TargetDirectory, # should be in relative form from root of repo. EG: sdk/servicebus
   $RootDirectory, # ideally $(Build.SourcesDirectory)
-  $VsoVariable = "" # target devops output variable
+  $AuthToken,
+  $VsoOwningUsers = "",
+  $VsoOwningTeams = "",
+  $VsoOwningLabels = ""
 )
+
+. (Join-Path ${PSScriptRoot} logging.ps1)
+
 $target = $TargetDirectory.ToLower().Trim("/")
 $codeOwnersLocation = Join-Path $RootDirectory -ChildPath ".github/CODEOWNERS"
 $ownedFolders = @{}
 
 if (!(Test-Path $codeOwnersLocation)) {
-  Write-Host "Unable to find CODEOWNERS file in target directory $RootDirectory"
+  LogWarning "Unable to find CODEOWNERS file in target directory $RootDirectory"
   exit 1
 }
 
-$codeOwnersContent = Get-Content $codeOwnersLocation
-
-foreach ($contentLine in $codeOwnersContent) {
-  if (-not $contentLine.StartsWith("#") -and $contentLine){
-    $splitLine = $contentLine -split "\s+"
-    
-    # CODEOWNERS file can also have labels present after the owner aliases
-    # gh aliases start with @ in codeowners. don't pass on to API calls
-    $ownedFolders[$splitLine[0].ToLower().Trim("/")] = ($splitLine[1..$($splitLine.Length)] `
-      | ? { $_.StartsWith("@") } `
-      | % { return $_.substring(1) }) -join ","
+function VerifyAlias($APIUrl)
+{
+  if ($AuthToken) 
+  {
+    $headers = @{
+      Authorization = "bearer $AuthToken"
+    }
+  }
+  try
+  {
+    $response = Invoke-RestMethod -Headers $headers $APIUrl
```

Comment:
How does this behave when `$headers` is null, since it's only specified in an if block?